### PR TITLE
feat: SC-13751: Compact IconButton style

### PR DESCRIPTION
**Pull request by bdow**

### CODE REVIEW

File: src/components/IconButton.tsx [hunk 1]
```diff
@@ -31,6 +33,7 @@ export function IconButton(props: IconButtonProps) {
     tooltip,
     menuTriggerProps,
     openInNew,
+    compact = false,
```
Comment (stephenh) [inline src/components/IconButton.tsx:36]: Should we check the presentation settings, i.e. icons in grid tables auto-compact? Or is that not desirable? I guess probably not...

**Review by stephenh — APPROVED** [review]

File: src/components/IconButton.tsx (hunk 1 above)
Comment (bdow) [reply src/components/IconButton.tsx:36]: Good question. I don't know of a case right now where the button size is dependent on the presentation settings or where its located. I'd add presentation setting support once we need it, or see it is necessary.

Comment (stephenh) [reply src/components/IconButton.tsx:36]: Yep, sounds good

**Review by aTrujilloHomeBound — APPROVED** [review]
great work!

**Pull request merged**